R/derive_var_atoxgr.R

```diff
@@ -0,0 +1,287 @@
+#' - NA is where a grade cannot be derived.
+#'
+#' @author Gordon Miller
+#'
+#' @return The input dataset with the character grade added
+#'
+#' @keywords der_bds_findings
+#'
+#' @family der_bds_findings
+#'
+#' @export
+#'
+#' @examples
+#' library(dplyr, warn.conflicts = FALSE)
+#'
+#' data <- tibble::tribble(
+#'   ~ATOXDSCL,                     ~AVAL,  ~ANRLO,   ~ANRHI, ~PARAM,
+#'   "Hypoglycemia",                119,    4,        7,      "Glucose (mmol/L)",
+#'   "Hypoglycemia",                120,    4,        7,      "Glucose (mmol/L)",
+#'   "Anemia",                      129,    120,      180,    "Hemoglobin (g/L)",
+#'   "White blood cell decreased",  10,     5,        20,     "White blood cell (10^9/L)",
+#'   "White blood cell decreased",  15,     5,        20,     "White blood cell (10^9/L)",
+#'   "Anemia",                      140,    120,      180,    "Hemoglobin (g/L)",
+#' )
+#'
+#' derive_var_atoxgr_dir(data,
+#'   new_var = ATOXGRL,
+#'   tox_description_var = ATOXDSCL,
+#'   meta_criteria = atoxgr_criteria_ctcv4,
+#'   criteria_direction = "L",
+#'   get_unit_expr = extract_unit(PARAM)
+#' )
+#'
+#' data <- tibble::tribble(
+#'   ~ATOXDSCH,                     ~AVAL,  ~ANRLO,   ~ANRHI, ~PARAM,
+#'   "Hyperglycemia",               119,    4,        7,      "Glucose (mmol/L)",
+#'   "Hyperglycemia",               120,    4,        7,      "Glucose  (mmol/L)",
+#'   "GGT increased",               129,    0,        30,     "Gamma Glutamyl Transferase (U/L)",
+#'   "Lymphocyte count increased",  4,      1,        4,      "Lymphocytes Abs (10^9/L)",
+#'   "Lymphocyte count increased",  2,      1,        4,      "Lymphocytes Abs (10^9/L)",
+#'   "GGT increased",               140,    120,      180,    "Gamma Glutamyl Transferase (U/L)",
+#' )
+#'
+#' derive_var_atoxgr_dir(data,
+#'   new_var = ATOXGRH,
+#'   tox_description_var = ATOXDSCH,
+#'   meta_criteria = atoxgr_criteria_ctcv4,
+#'   criteria_direction = "H",
+#'   get_unit_expr = extract_unit(PARAM)
+#' )
+derive_var_atoxgr_dir <- function(dataset,
+                                  new_var,
+                                  tox_description_var,
+                                  meta_criteria = atoxgr_criteria_ctcv4,
+                                  criteria_direction,
+                                  get_unit_expr) {
+  new_var <- assert_symbol(enquo(new_var))
+  tox_description_var <- assert_symbol(enquo(tox_description_var))
+  get_unit_expr <- assert_expr(enquo(get_unit_expr))
+
+  # Check Grade description variable exists on input data set
+  assert_data_frame(dataset, required_vars = vars(!!tox_description_var))
+
+  # Check metadata data set has required variables
+  assert_data_frame(
+    meta_criteria,
+    required_vars = vars(TERM, GRADE_CRITERIA_CODE, DIRECTION, SI_UNIT_CHECK, VAR_CHECK)
+  )
+
+  # check input parameter has correct value
+  assert_character_scalar(criteria_direction, values = c("L", "H"))
```

Comment:
I think it would be good to also incorporate checks for the metadata `DIRECTION` column to make sure the values are "L" and "H".
Minor: also maybe move this line up so it's together with the other checks?

Reply:
added extra check in, and moved check further up. Good suggestion.